BlueprintUI/Sources/Layout/AspectRatio.swift

```diff
@@ -0,0 +1,29 @@
+/// Represents an a proportional relationship between width and height.
+public struct AspectRatio {
+    /// A 1:1 aspect ratio.
+    public static let square = AspectRatio(x: 1, y: 1)
+
+    /// The horizontal component of this ratio.
+    public var x: CGFloat
```

Comment:
The property names `x`/`y` imply a position in a coordinate system – if this structure persists I would almost certainly rename them to `width`/`height`.

Stepping back, I find it surprising that we are storing two values here. Can we simply store the aspect ratio (derived from `width/height`) as the backing storage for this type? Convenience initializers that take `width`/`height` or a `CGSize` seem useful, but it seems as though this type should encode the most fundamental representation of an aspect ratio as its storage: a single `Double` value.

Thoughts?

Reply:
For API reference, SwiftUI has the following:

```swift
extension View {
    public func aspectRatio(_ aspectRatio: CGFloat? = nil, contentMode: ContentMode) -> some View
    public func aspectRatio(_ aspectRatio: CGSize, contentMode: ContentMode) -> some View
}

public enum ContentMode {
    case fit
    case fill
}
```

Reply:
Agree `width`/`height` is better, I kind of punted on that (I did check Wikipedia to see if they had proper names first, but they don't seem to).

Here's an argument for storing two values:
- You get better diagnostics. `AspectRatio(width: 16, height: 9)` can have a debug string of `16:9` instead of `1.7777777777778`.
- A bit more resilient to transforms. If you have an existing ratio and you want to double its height, `aspectRatio.height *= 2` doesn't lose precision to normalization.

I recognize that's a pretty weak argument, and I'd rather match SwiftUI, so I'll change this.